### PR TITLE
done with bonuses🥳

### PR DESCRIPTION
**Pull request by habidbesp**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const express = require('express')
+
+const app = express()
+
+app.use(cors())
+
+
+app.get("/json", (request, response) => {
+    // Send the whole modules array as JSON to the browser
+    console.log("you basterd want to set up a route, right?");
```
Comment (losrobbos) [inline server.js:17]: Yes, I want :)

**Comment by habidbesp** [comment]
Jajaja, just learning from my teacher 😂

File: server.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Define an array with the DCI course modules at the top of the file:
+
+const moduleDCI = ['HTML/CSS', 'JavaScript', 'DOM', 'SPA', 'Backend', 'Fullstack', 'Final Project']
+
+const { request, response } = require('express');
```
Comment (losrobbos) [inline server.js:5]: These imports are actually not necessary. But maybe it was imported by your VSCode somehow accidentally :)

File: server.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+app.get("/json", (request, response) => {
+    // Send the whole modules array as JSON to the browser
+    console.log("you basterd want to set up a route, right?");
+    response.json(moduleDCI)
+})
+
+// Bonus Task 1
+
+const textStg = moduleDCI.join(', ')
```
Comment (losrobbos) [inline server.js:23]: That line here I would move into the corresponding route /text, then al logic belonging to that route is at the right place :)

File: server.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+app.get("/text", (request, response) => {
+    console.log("this is the 'text' route 'mdf'");
+    response.json(textStg)
+})
+
+
+const liItems = moduleDCI.map(list => {
```
Comment (losrobbos) [inline server.js:31]: Same here, this map, even though fantastic :), should get stuffed into the /html route

File: server.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+app.get("/html", (request, response) => {
+    console.log("this is the route 'html'");
+    response.send(ul)
+})
+
+
+
+app.listen(5000, () => {
+    // Provide a success message that the server started up
+    console.log('API has started successfully on PORT 5000 💪');
```
Comment (losrobbos) [inline server.js:51]: Nice emoticon :D

**Comment by losrobbos** [comment]
Overall all good. In the backend there were just minor things with code organisation, but nothing serious.

The React part is impressive, even with Context and complete application structure. Cool stuff :+1: :100:

**Pull request closed without merging**